src/compiler/config/test/load-config.spec.ts

```diff
@@ -1,26 +1,53 @@
+import { getMockFSPatch } from '@stencil/core/testing';
+import { createNodeSys } from '@sys-api-node';
+import fs from 'fs';
+import mock from 'mock-fs';
 import path from 'path';
 
 import { ConfigFlags } from '../../../cli/config-flags';
-import { createSystem } from '../../../compiler/sys/stencil-sys';
 import type * as d from '../../../declarations';
 import { normalizePath } from '../../../utils';
 import { loadConfig } from '../load-config';
 
 describe('load config', () => {
-  const configPath = require.resolve('./fixtures/stencil.config.ts');
-  const configPath2 = require.resolve('./fixtures/stencil.config2.ts');
-  const fixturesPath = path.dirname(configPath);
-  const srcPath = path.join(fixturesPath, 'src');
-  const indexPath = path.join(srcPath, 'index.ts');
+  const configPath = 'fixtures/stencil.config.ts';
+  const configPath2 = 'fixtures/stencil.config2.ts';
   let sys: d.CompilerSystem;
+  const noTsConfigPath = 'no-ts-dir/stencil.config.ts';
 
   beforeEach(() => {
-    sys = createSystem();
-    sys.writeFileSync(configPath, ``);
-    sys.writeFileSync(configPath2, ``);
-    sys.createDirSync(fixturesPath);
-    sys.createDirSync(srcPath);
-    sys.writeFileSync(indexPath, `console.log('fixture');`);
+    sys = createNodeSys();
+
+    const tsconfig = JSON.stringify(
+      {
+        compilerOptions: {
+          allowSyntheticDefaultImports: true,
+          experimentalDecorators: true,
+          lib: ['dom', 'es2015'],
+          moduleResolution: 'node',
+          module: 'esnext',
+          target: 'es2017',
+          jsx: 'react',
+          jsxFactory: 'h',
+          jsxFragmentFactory: 'Fragment',
+        },
+        include: ['src'],
+      },
+      null,
+      2
+    );
+
+    mock({
+      [configPath]: mock.load(path.resolve(__dirname, configPath)),
+      [configPath2]: mock.load(path.resolve(__dirname, configPath2)),
+      'fixtures/tsconfig.json': tsconfig,
+      [noTsConfigPath]: mock.load(path.resolve(__dirname, configPath)),
+      ...getMockFSPatch(mock),
```

Comment:
Here we create files in the in-memory mock filesystem used by `mock-fs`. the `mock.load` basically acts as a pass-through to the actual filesystem, so this lets us use things in the `./fixtures` directory without having to worry about any fs writes in this test making it through to the actual disk.

Reply:
That's good to know about `mock.load` might be worth adding a comment briefly explaining that